routes/documents.py

```diff
@@ -170,6 +171,8 @@ def submit_document():
     file = request.files['file']
     if not _allowed_file(file.filename):
         raise ClientError('file extension not allowed', status=406)
+    if data['validated'] and not current_user.is_authenticated:
```

Comment:
`validated` is optional, so marshmallow won't throw any errors if it's not there, yet you try to access it here.
This breaks regular document submission (and is generally unsafe).